File: 0101-0200/0153.Find_Minimum_in_Rotated_Sorted_Array.java
//0ms



class Solution {
    public int findMin(int[] nums) {
        int start = 0;
        int end = nums.length - 1;
        int ans = Integer.MAX_VALUE;
        while (start <= end) {
            int mid = start + (end - start) / 2;
            if (nums[start] <= nums[end]) {
                // It's already in sorted order. Then the first element is the minimum.
                ans = Math.min(ans, nums[start]);
                break;
            }

            if (nums[start] <= nums[mid]) {
                // The start element is the minimum of the entire left half. Store and move to the right.
                ans = Math.min(ans, nums[start]);
                start = mid + 1;
            } else {
                // Possible that we are in two halves of rotated array. Store and move left.
                ans = Math.min(ans, nums[mid]);
                end = mid - 1;
            }
        }
        return ans;
    }
}






//0ms






class Solution {
    public int findMin(int[] nums) {
        int start = 0;
        int end = nums.length-1;
         
    if(nums[0]<=nums[nums.length-1]){
                 return nums[0];
          }
        while(start<end){
            int mid= (start+end)/2;
            if(nums[mid]>nums[mid+1]){
                return nums[mid+1];
            }
            if(nums[mid]>nums[start]){
                start= mid+1;
            }else{
                end=mid;
            }
        }
        return nums[start];
        
    }
}

File: 0101-0200/0153.Find_Minimum_in_Rotated_Sorted_Array.js
//0ms




/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    let left = 0, right = nums.length - 1;
    while (left < right) {
        let mid = Math.floor(left + (right - left) / 2);
        
        if (nums[mid] < nums[right]) {
            right = mid;
        } else {
            left = mid + 1;
        }
    }
    return nums[left];
};



//0ms




/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    return Math.min(...nums)
};
